mantidproject/mantid
# Indirect Settings GUI

**Description of work.**
This PR introduces an Indirect 'Settings' GUI which can be used to store and change how commonly shared functionality works on the Indirect interfaces.

An important option is the choice to turn off the filtering of input data by the name of the data. This will be a useful feature for users at the `ILL`, and even for some users at `ISIS`.
Another option that can be chosen is which facility is selected. The tabs shown on the Reduction interfaces depends on the facility so this will make it easier to quickly switch between facilities.

Other options may also be added in the future such as plotting options where you can turn off or on the plotting of error bars for output data on all interfaces.

**To test:**
1.`Interfaces`->`Indirect`->`Data Analysis` or `Data Reduction` or `Data Corrections` or `Bayes`
2. Click on the settings button (bottom left next to the help ? button)
3. There should be three options. 
- To change the facility to the relevant facility ISIS, ILL, SNS or MLZ.
- To restrict input data by name
- To turn on or off the plotting of error bars for the output data

4. Uncheck the `Restrict input data by name` option and click `Ok`
5. Attempt to load data with a different naming convention than is usually accepted.
6. Check the `Restrict input data by name` option.
7. Make sure that input data is restricted by their name.

8. A good test for the facilities being applied correctly would be on the Data Reduction interface. On this interface change the facility and you should see that the number of tabs seen on the interface will vary depending on the facility selected.

9. Turn on and off the `plot error bars` option and check that error bars are plotted at the appropriate time for the output options. Note that the error bars are only changed for the output options.

Fixes #24735

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
